modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -737,27 +738,45 @@ public String toString() {
         }
     }
 
+    // visible for testing
+    record StatsCounter(LongAdder operations, LongAdder requests) {
+        StatsCounter() {
+            this(new LongAdder(), new LongAdder());
+        }
+
+        EndpointStats getEndpointStats() {
+            return new EndpointStats(operations.sum(), requests.sum());
+        }
+
+        EndpointStats addTo(EndpointStats other) {
+            long ops = operations.sum() + other.operations();
+            long reqs = requests.sum() + other.requests();
+            return new EndpointStats(ops, reqs);
+        }
```

Comment:
Ah it was just slightly lighter-weight, so rather than doing
```java
return new EndpointStats(operations.sum(), requests.sum()).add(other);
```
which would create two `EndpointStats` just to do the addition (seemed excessive to just add some longs), I figured this would just create a single new one with the new values. It may be premature optimisation. At very least I should be doing `addExact` here.